packages/flame_forge2d/lib/contact_callbacks.dart

```diff
@@ -1,93 +1,55 @@
-  bool inOrder(Object o1, Object o2) => o1 is T1 && o2 is T2;
-
-  // Remember that this is not symmetric, it checks if the types in `o1` and
-  // `o2` are the same or inherits from the types in `other`
-  bool match(Object o1, Object o2) =>
-      (o1 is T1 && o2 is T2) || (o2 is T1 && o1 is T2);
-}
-
-typedef ContactCallbackFun = void Function(Object a, Object b, Contact contact);
-
-abstract class ContactCallback<Type1, Type2> {
-  ContactTypes<Type1, Type2> types = ContactTypes<Type1, Type2>();
-
-  void begin(Type1 a, Type2 b, Contact contact) {}
-  void end(Type1 a, Type2 b, Contact contact) {}
-  void preSolve(Type1 a, Type2 b, Contact contact, Manifold oldManifold) {}
-  void postSolve(Type1 a, Type2 b, Contact contact, ContactImpulse impulse) {}
-}
-
-class ContactCallbacks extends ContactListener {
-  final List<ContactCallback> _callbacks = [];
-
-  void register(ContactCallback callback) {
-    _callbacks.add(callback);
+import 'flame_forge2d.dart';
+
+/// Used to listens to a [BodyComponent] contacts events.
+///
+/// Contacts events occur whenever two [Fixture] meet each other.
+///
+/// To react to contacts you should assign [ContactCallbacks] to a [Body]'s
+/// userData or/and to [Fixture]'s userData.
+/// {@macro flame_forge2d.world_contact_listener.algorithm}
+class ContactCallbacks {
+  /// Called when two [Fixture]s begin to overlap.
+  ///
+  /// It is called for sensors and non-sensors.
+  void beginContact(Object other, Contact contact) {
+    onBeginContact?.call(other, contact);
   }
 
-  void deregister(ContactCallback callback) {
-    _callbacks.remove(callback);
+  /// Called when two [Fixture]s cease to overlap.
+  ///
+  /// It is called for sensors and non-sensors.
+  ///
+  /// It may be called when a body is destroyed.
+  void endContact(Object other, Contact contact) {
+    onEndContact?.call(other, contact);
   }
 
-  void clear() {
-    _callbacks.clear();
+  /// Called after collision detection, but before collision resolution.
+  ///
+  /// This gives you a chance to disable the [Contact] based on the current
+  /// configuration.
+  ///
+  /// Sensors do not create [Manifold]s.
+  void preSolve(Object other, Contact contact, Manifold oldManifold) {
+    onPreSolve?.call(other, contact, oldManifold);
   }
 
-  void _maybeCallback(
-    Contact contact,
-    ContactCallback callback,
-    ContactCallbackFun f,
-  ) {
-    final a = contact.fixtureA.body.userData;
-    final b = contact.fixtureB.body.userData;
-    final wanted = callback.types;
-
-    if (a == null || b == null) {
-      return;
-    }
-
-    if (wanted.match(a, b) ||
-        (wanted.has(BodyComponent) && wanted.hasOne(a, b))) {
-      wanted.inOrder(a, b) ? f(a, b, contact) : f(b, a, contact);
-    }
+  /// Called after collision resolution.
+  ///
+  /// Usually defined to gather collision impulse results.
+  ///
+  /// If one of the colliding objects is a sensor, this will not be called.
```

Comment:
```suggestion
  /// Usually defined to gather collision impulse results.
  /// If one of the colliding objects is a sensor, this will not be called.
```

Reply:
I'm against this change since it goes against Effective Dart guidelines.

Refer to: https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

Reply:
There is a single sentence before the paragraph starts, it's easy to miss it when only looking at the suggested change.